pkg/reaper/datacenter.go

```diff
@@ -7,43 +7,44 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-func AddReaperSettingsToDcConfig(reaperTemplate *reaperapi.ReaperClusterTemplate, dcConfig *cassandra.DatacenterConfig) {
-	addUser(reaperTemplate, dcConfig)
+func AddReaperSettingsToDcConfig(reaperTemplate *reaperapi.ReaperClusterTemplate, dcConfig *cassandra.DatacenterConfig, authEnabled bool) {
+	if authEnabled {
+		addUser(reaperTemplate, dcConfig)
```

Comment:
I wonder: if the user specified a CQL user secret in the Reaper CRD, but auth is disabled globally, should we go ahead and create that user?

Reply:
One thing to keep in mind: if the secret exists, the Reaper controller will add an extra `REAPER_CASS_AUTH_ENABLED=true` env var to the Reaper pods. This is likely to make Reaper startup fail if auth is disabled in Cassandra.

Reply:
Good question. I don't have a strong opinion one way or the other at this point. This brings up another question though. If the user has auth enabled and then subsequently disables it, should delete any secrets and users that we create?

Reply:
Indeed, I haven't thought about that. I guess we should delete them if they were created by us, otherwise leave them be (as orphaned resources) if they were user-created. I'll add this item to my TODO list, not sure I'll have time to tackle it today.

Reply:
> If the user has auth enabled and then subsequently disables it, should delete any secrets and users that we create?

@jsanda I eventually decided to not delete them. Mainly because afaict the replication controller is not able to detect that a `corev1.Secret` object was deleted and that it should also be deleted from other data planes.

So currently, if auth was enabled and is subsequently disabled:

* If Reaper secrets were created manually by the user, they should have the Orphan annotation. Thus they won't ever be deleted, even if the `K8ssandraCluster` CRD is deleted.
* If Reaper secrets were generated by k8ssandra controller, they won't have the Orphan annotation. They won't be deleted when auth is disabled, however they will get deleted if the `K8ssandraCluster` CRD is deleted.

Does that sound OK to you?

I put a TODO here just in case: https://github.com/k8ssandra/k8ssandra-operator/blob/f221f13604589355cabe14c50e2d375aedc81470/controllers/k8ssandra/secrets.go#L60-L61

Reply:
Also note that the Cassandra superuser secret is always created and replicated, regardless of authentication.

I explained why here: https://github.com/k8ssandra/k8ssandra-operator/blob/f221f13604589355cabe14c50e2d375aedc81470/controllers/k8ssandra/secrets.go#L14-L19